neonvm-daemon/pkg/cpuscaling/sys_fs_state_scaling.go

```diff
@@ -0,0 +1,196 @@
+package cpuscaling
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// CPU directory path
+const cpuPath = "/sys/devices/system/cpu/"
+
+type CPUSysFsStateScaler struct {
+}
+
+func (c *CPUSysFsStateScaler) EnsureOnlineCPUs(X int) error {
+	cpus, err := getAllCPUs()
+	if err != nil {
+		return err
+	}
+
+	onlineCount, err := c.GetActiveCPUsCount()
+	if err != nil {
+		return err
+	}
+
+	if onlineCount < uint32(X) {
+		for _, cpu := range cpus {
+			if cpu == 0 {
```

Comment:
nit: this is a lot of nested control flow. Maybe worth separating into >1 function?